hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/HoodieTimelineArchiver.java

```diff
@@ -479,6 +479,8 @@ private Stream<HoodieInstant> getCommitInstantsToArchive() throws IOException {
   }
 
   private Stream<HoodieInstant> getInstantsToArchive() throws IOException {
+    metaClient.reloadActiveTimeline();
+
```

Comment:
The `metaClient` comes from the hoodie table passed in from the constructor, the table should be a fresh new one, so what is the problem here ?

Reply:
![image](https://user-images.githubusercontent.com/37108074/211502949-d4397c9e-e6c8-414b-9f73-7494c4d08096.png)
After commit finished, the instant in table is not updated (still INFLIGHT), therefore, autoArchiveOnCommit in postCommit will not include the just written instant.
Maybe we can reload it inside the commit, like this:
![image](https://user-images.githubusercontent.com/37108074/211503701-aae49587-71cd-4cef-9d8f-016caf119905.png)

Reply:
Good catch, what is actualy the effect then? The instants would be sill archived for the next commit right?

Reply:
@danny0405 just the results can be confusing, like set `hoodie.keep.max.commits=5` , but after 10 commits, there will be 6 commits (more than 5) in active timeline.
And like you say, they would be archived after the next commit